### PR TITLE
Maryna/feature/ticket order

### PR DESCRIPTION
**Pull request by marynag**

### CODE REVIEW

File: app/forms/ticket_order.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from wtforms import StringField, IntegerField, ValidationError
+from wtforms.validators import DataRequired
+
+from app import models as m
+from app import db
+
+
+class TicketOrderForm(FlaskForm):
+    amount = IntegerField("amount", [DataRequired()])
+    message = StringField("message")
```
Comment (kipod) [inline app/forms/ticket_order.py]: Message for ticket order? Hmmm...

File: app/forms/ticket_order.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ticket_time_id = IntegerField("ticket_time_id", [DataRequired()])
+    ticket_date_id = IntegerField("ticket_date_id", [DataRequired()])
+    current_user_id = IntegerField("current_user_id", [DataRequired()])
```
Comment (kipod) [inline app/forms/ticket_order.py]: current_user_id - please remove

Comment (kipod) [reply app/forms/ticket_order.py]: the current user is always well known...

File: app/models/ticket_order.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    ticket_date_id: orm.Mapped[int] = orm.mapped_column(
+        sa.Integer, sa.ForeignKey("tickets_date.id"), nullable=False
+    )
+    ticket_time_id: orm.Mapped[int] = orm.mapped_column(
+        sa.Integer, sa.ForeignKey("tickets_time.id"), nullable=False
+    )
+    amount: orm.Mapped[int] = orm.mapped_column(sa.Integer, nullable=False)
+    message: orm.Mapped[str] = orm.mapped_column(sa.String(127), default="")
+    current_user_id: orm.Mapped[int] = orm.mapped_column(sa.Integer, nullable=False)
```
Comment (kipod) [inline app/models/ticket_order.py]:
```python
user_id: orm.Mapped[int] = orm.mapped_column(
        sa.Integer, sa.ForeignKey("users.id"), nullable=False
    )
```

File: app/models/ticket_order.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    status: orm.Mapped[s.Status] = orm.mapped_column(
+        sa.Enum(s.Status),
```
Comment (kipod) [inline app/models/ticket_order.py]: s.Status --> s.OrderStatus

File: app/models/ticket_order.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        default=s.Status.IN_PROGRESS,
```
Comment (kipod) [inline app/models/ticket_order.py]: default=NEW

File: app/schema/order_status.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import enum
+
+
+class Status(enum.Enum):
+    IN_PROGRESS: str = "IN PROGRESS"
+    DONE: str = "DONE"
+    CANCELED: str = "CANCELED"
```
Comment (kipod) [inline app/schema/order_status.py]:
```python
class Status(enum.Enum):
    NEW = "NEW"
    IN_PROGRESS = "IN PROGRESS"
    DONE = "DONE"
    CANCELED = "CANCELED"

Comment (kipod) [reply app/schema/order_status.py]: first value - NEW - is default value
IN_PROGRESS will be set by bot when it be taken in process

File: app/views/ticket_order.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+bp = Blueprint("ticket_order", __name__, url_prefix="/order")
+
+
+@bp.route("/create", methods=["POST"])
+@login_required
+def create():
+    form = f.TicketOrderForm()
+    if form.validate_on_submit():
+        order = m.TicketOrder(
+            current_user_id=form.current_user_id.data,
```
Comment (kipod) [inline app/views/ticket_order.py]: ```current_user``` --->> ```from flask_login import current_user```

than...

```python
current_user_id=current_user.id,
```

File: app/views/ticket_order.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            ticket_time_id=form.ticket_time_id.data,
+            ticket_date_id=form.ticket_date_id.data,
+            amount=form.amount.data,
+        )
+        log(log.INFO, "Ticket order submitted. Order: [%s]", order)
+        flash("Ticket order added!", "success")
+
+        order.save()
+    else:
+        log(log.INFO, "Ticket order form validation failed. Form: [%s]", form)
```
Comment (kipod) [inline app/views/ticket_order.py]: INFO --> ERROR

**Review by kipod — COMMENTED** [review]
Thanks! Great job.

Please see my comments

File: app/forms/ticket_order.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+from app import models as m
+from app import db
+
+
+class TicketOrderForm(FlaskForm):
+    amount = IntegerField("amount", [DataRequired()])
+    ticket_time_id = IntegerField("ticket_time_id", [DataRequired()])
+    ticket_date_id = IntegerField("ticket_date_id", [DataRequired()])
+    user_id = IntegerField("user_id", [DataRequired()])
```
Comment (nixissmtp) [inline app/forms/ticket_order.py]: remove user_id

File: app/templates/ticket/add.html [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        <div class="p-4 border-b rounded-t dark:border-gray-600 mb-4 flex flex-col gap-2"> 
+            <input type="hidden" id="dateHiddenInput" name="ticket_date_id" />
+            <div class="text-xl font-medium text-gray-500 dark:text-gray-400">
+                Floor: <span id="floorNamePlaceholder"></span>
+            </div>
+            <div class="text-gray-900 dark:text-white text-xl font-semibold">
+                Time: <span id="timePlaceholder" class="text-2xl"></span> <input type="hidden" id="timeHiddenInput" name="ticket_time_id" />               
+            </div>                        
+            <div class="text-xl font-medium text-gray-500 dark:text-gray-400">
+                User: {{current_user.username}} <input type="hidden" name="current_user_id" value="{{current_user.id}}"/>
```
Comment (nixissmtp) [inline app/templates/ticket/add.html]: remove hidden input

**Review by nixissmtp — APPROVED** [review]

**Pull request merged**